webapp/graphite/finders/standard.py

```diff
@@ -80,8 +82,15 @@ def _find_paths(self, current_dir, patterns):
     else:
       entries = [ pattern ]
 
-    subdirs = [entry for entry in entries if isdir(join(current_dir, entry))]
-    matching_subdirs = match_entries(subdirs, pattern)
+    if using_globstar:
+        matching_subdirs = map(operator.itemgetter(0), os.walk(current_dir))
```

Comment:
I tested independently and only saw directories show up.

Output is only directories:
```
$ python2.7 test.py | xargs -L1 -I{}  stat -c%F {} | uniq -c
  27177 directory
```

It did take 1.1 seconds to just run the osdir on this host, but it is at about 65% IO utilization all the time.
```
$ time python2.7 test.py  | wc -l
27177

real	0m1.149s
user	0m0.675s
sys	0m0.478s
```

Test script:
```
$ cat test.py
import os
import operator

current_dir='/opt/graphite/storage/whisper/carbon/'

matching_subdirs = map(operator.itemgetter(0), os.walk(current_dir))

for i in matching_subdirs:
  print i
```